src/client/cockpit-client

```diff
@@ -163,6 +184,45 @@ class ExternalCockpitWs:
         pass
 
 
+class InternalCockpitWs:
+    def start(self):
+        self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.socket.bind(('127.0.90.90', 0))
+        self.socket.listen()
+
+        self.write_config()
+        self.ws = subprocess.Popen([f'{libexecdir}/cockpit-ws'],
+                                   preexec_fn=self.preexec_fn, pass_fds=[3])
+
+        host, port = self.socket.getsockname()
+        return f'http://{host}:{port}/'
+
+    def preexec_fn(self):
+        os.environ['XDG_CONFIG_DIRS'] = self.config_dir.name
+        os.environ['LISTEN_PID'] = str(os.getpid())
+        os.environ['LISTEN_FDS'] = str(1)
+        prctl(prctl.SET_PDEATHSIG, signal.SIGKILL)
```

Comment:
OOI, why this big hammer? TERM not good enough?

Reply:
To turn it around: would TERM help anything?  This is a stateless process and we want it to simply go away.  Using TERM seems like a good way to end up with a weird edge case where we have a cockpit-ws hanging around after the fact.

Reply:
Good point.